strawberryfields/gbs/plot.py

```diff
@@ -0,0 +1,235 @@
+web_green2 = '#3e9651'
+web_red1 = '#d35e60'
+web_red2 = '#cc2529'
+sf_cyan = '#3e9bb7'
+sf_cyan_light = '#6eb4c9'
+sf_lightgreen = '#daf0e9'
+sf_red = '#d35e60'
+sf_grey = '#efefef'
+light_grey = '#CDCDCD'
+
+graph_node_colour = web_green2
+graph_edge_colour = light_grey
+subgraph_node_colour = web_red2
+subgraph_edge_colour = web_red2
+
+graph_node_size = 14
+subgraph_node_size = 16
+
+
+def plot(graph: nx.Graph, subgraph: Optional[list] = None, size: float = 500) -> None:
+    """ Creates a plot.ly plot of the input graph.
+
+    The graph layout is fixed to be the Kamada-Kawai layout with an aspect ratio of 1:1. The
+    function can plot just the input graph or the graph with a specified subgraph highlighted.
+    The function uses the standard colour theme of green nodes, grey edges, and red highlighted
+    subgraph.
+
+    **Example usage**:
+        >>> graph = nx.complete_graph(10)
+        >>> fig = plot(graph, [0, 1, 2, 3])
+        >>> fig.show()
+
+        Args:
+            graph (nx.Graph): input graph
+            subgraph (list): list of nodes comprising the subgraph to highlight
+            size (dict): size of the plot
+
+        Returns:
+             plot.ly graph object
+        """
+
+    s = graph.subgraph(subgraph)
+    l = nx.kamada_kawai_layout(graph)
+
+    g_nodes = go.Scatter(
+        **_node_coords(graph, l),
+        mode='markers',
+        hoverinfo='text',
+        marker=dict(color=graph_node_colour, size=graph_node_size, line_width=2)
+    )
+
+    g_edges = go.Scatter(
+        **edge_coords(graph, l),
+        line=dict(width=1, color=graph_edge_colour),
+        hoverinfo='none',
+        mode='lines'
+    )
+
+    g_nodes.text = [str(i) for i in graph.nodes()]
+
+    layout = go.Layout(showlegend=False, hovermode='closest',
+                       xaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
+                       yaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
+                       margin=dict(b=0, l=0, r=0, t=25),
+                       height=size,
+                       width=size,
+                       plot_bgcolor='#ffffff'
+                       )
+
+    if subgraph:
+
+        s_edges = go.Scatter(
+            **edge_coords(s, l),
+            line=dict(width=2, color=subgraph_edge_colour),
+            hoverinfo='none',
+            mode='lines'
+        )
+
+        s_nodes = go.Scatter(
+            **_node_coords(s, l),
+            mode='markers',
+            hoverinfo='text',
+            marker=dict(color=subgraph_node_colour, size=subgraph_node_size, line_width=2)
+        )
+
+        s_nodes.text = [str(i) for i in s.nodes()]
+
+        f = go.Figure(data=[g_edges, s_edges, g_nodes, s_nodes], layout=layout)
+
+    else:
+        f = go.Figure(data=[g_edges, g_nodes], layout=layout)
+
+    return f
+
+
+def plot_subgraph(subgraph: nx.Graph, size: float = 500) -> None:
+    """ Creates a plot.ly plot of the input subgraph.
```

Comment:
```suggestion
    """Creates a plot.ly plot of the input subgraph.
```

Reply:
We should decide on Plotly vs plot.ly (which I am kind of neutral aside from desire for consistency)

Reply:
I actually prefer just plotly, so let's go for that?